src/commands.jl

```diff
@@ -272,6 +272,22 @@ function maybe_reset_frame!(@nospecialize(recurse), frame::Frame, @nospecialize(
     return frame, pc
 end
 
+# Unwind the stack until an exc is eventually caught, thereby
+# returning the frame that caught the exception at the pc of the catch
+# or rethrow the error
+function unwind_exception(frame::Frame, exc)
+    while frame !== nothing
+        if !isempty(frame.framedata.exception_frames)
+            # Exception caught
+            frame.pc = frame.framedata.exception_frames[end]
+            frame.framedata.last_exception[] = exc
+            return frame
+        end
+        frame = frame.caller
```

Comment:
Should we set `frame.callee = nothing`? If it gets caught and execution continues, that `@assert` in `step_expr!` will be trouble.

Reply:
Also could probably `recycle`? Do we need to check `break_on_err`? This is starting to feel a bit like `handle_err`.

Reply:
`break_on_error` would have made this not throw in the first place, no? It would have looked down the stack, found that something would catch the exception and then instead returned breakpoint.

Yes, setting the `callee` is needed.